packages/transformers/svelte/src/SvelteTransformer.js

```diff
@@ -0,0 +1,147 @@
+import {Transformer} from '@parcel/plugin';
+import SourceMap from '@parcel/source-map';
+import {relativeUrl} from '@parcel/utils';
+import {compile, preprocess} from 'svelte/compiler';
+import ThrowableDiagnostic from '@parcel/diagnostic';
+
+export default new Transformer({
+  async loadConfig({config, options, logger}) {
+    const svelteConfig = await config.getConfig([
+      '.svelterc',
+      'svelte.config.js',
+    ]);
+    if (!svelteConfig) return {};
+    if (svelteConfig.filePath.endsWith('.js')) {
+      // TODO: Is there a better way of handling this warning? Probably just
+      // mention it in the documentation and silently invalidate.
+      logger.warn({
+        message:
+          'WARNING: Using a JavaScript Svelte config file means losing ' +
+          'out on caching features of Parcel. Use a .svelterc(.json) ' +
+          'file whenever possible.',
+      });
+      config.invalidateOnStartup();
+    }
+    return {
+      ...svelteConfig.contents,
+      compilerOptions: {
+        // TODO: Call this out in the documentation.
+        css: false,
+        ...svelteConfig.contents.compilerOptions,
+        dev: options.mode !== 'production',
+      },
+    };
+  },
+
+  async transform({asset, config, options, logger}) {
+    let source = await asset.getCode();
+    const filename = relativeUrl(options.projectRoot, asset.filePath);
+
+    if (!config.preprocess) {
+      let preprocessor = null;
+      try {
+        // This seems sufficiently batteries included. Only apply
+        // svelte-preprocess if it is importable. Otherwise, assume the lack of
+        // a preprocess entry is intentional.
+        // TODO: Is it possible to utilize parcel's pipelines in lieu of
+        // svelte-preprocess?
```

Comment:
The Vue transformer uses named pipelines to first extract the individual assets from the input file and then process the parts again.

Reply:
So, if I'm understanding the Vue transformer correctly, it transforms the vue code once to produce some "glue" code which imports the disparate named pipelines. Those named pipelines get processes by the Vue transformer separately and each get rendered into the glue code. My use case would be to transform the embed code as a callback within the preprocess function.

I don't think I want to take up that change in this PR. It's probably better to keep it simple. It seems pretty likely to go wrong with svelte, if I were to follow the same pattern as Vue by generating glue code. It would also be unexpected from a UX perspective.

Reply:
It might be weird if TypeScript/SASS/PostCSS behave differently in a Svelte file than outside...

We used to have a feature that allowed returning multiple assets for processing, and then accessing them in a `postProcess` method. It was originally designed for cases like this, but we removed it because no one was using it. #5674

Reply:
I think it's actually the opposite. As it stands right now, preprocessing is optional and only occurs when the user has explicitly set the `preprocess` key in the `svelte.config.js` or `svelte-preprocess` is importable. In the first case, the user is fully in control of what and how the embed code gets processed. In the second case, `svelte-preprocess` [tells](https://github.com/sveltejs/svelte-preprocess/blob/main/docs/getting-started.md) the user to install the tsc compiler. Perhaps a good solution would be to document the inconsistency and prompt the user to install the tsc transformer if using svelte?

As to the `postProcess` function, I would love to use this if the project is willing to maintain the API and if it works for this use case. I don't think I can commit to that maintenance though as it would likely require deeper knowledge of the inner workings of parcel. I have concerns whether it would actually work for this situation though. Would I be able to fully run an embed pipeline to completion within the transform function? `svelte/compiler:preprocess` takes as input callbacks to preprocess the embed languages of a svelte file. The output of those callbacks need to be vanilla js, css, or markup. I imagine optimization would also impact that. I don't think I would be able to take the multi-pass approach that Vue took.

The most conservative solution would be for me to remove preprocessing altogether so there's no backwards incompatible changes in the defaults at any point. Though I think the current defaults requiring the user to opt into `svelte-preprocess` is sufficient.